docs/dev/clusteroperator.md

```diff
@@ -64,9 +64,15 @@ Refer [the godocs](https://godoc.org/github.com/openshift/api/config/v1#ClusterS
 
 In general, ClusterOperators should contain at least three core conditions:
 
-* `Progressing` which is true if the controller has detected the state is invalid and is working towards it (or if the controller has never reconciled before)
-* `Available` should be true if the controller has successfully reached a working status and no known errors are blocking progress.
-* `Failing` should be true if the controller is blocked from reaching the desired state.
+* `Progressing` must be true if the operator is actually making change to the binary maintained by the operator (operand).
+The change may be anything: desired user state, desired user configuration, observed configuration, version update, etc.
+If this is false, it means the operator is not trying to apply any new state.
+If it remains true for an extended period of time, it suggests something is wrong in the cluster.  It can probably wait until Monday.
+* `Available` must be true if the binary maintained by the operator (operand) is functional and available in the cluster.
+If this is false, it means there is an outage.  Someone is probably getting paged.
+* `Failing` should be true if the operator has encountered an error that is preventing it or it's operand from working properly.
+The operand may still be available, but intent may not have been fulfilled.
+If this is true, it means that the operand is at risk of an outage or improper configuration.  It can probably wait until the morning, but someone needs to look at it.
```

Comment:
i don’t know of a case where available false and failing not true

Reply:
> i don’t know of a case where available false and failing not true

1. initial creation
2. no nodes available
3. controller manager down, so the operator created a workload resource and is waiting for pods
4. apiservice for operand is Available=false, but there was no error creating it.

Reply:
These are good examples to have in the doc